Mlem/Views/Shared/Moderation/Tools/BanUserView+Logic.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  BanUserView+Logic.swift
+//  Mlem
+//
+//  Created by Sjmarf on 09/03/2024.
+//
+
+import SwiftUI
+
+extension BanUserView {
+    func confirm() {
+        if !banFromInstance, let communityContext {
+            communityBan(from: communityContext)
+        } else {
+            instanceBan()
+        }
+    }
```

Comment:
*nit*

Technically this could lead to an instance ban being sent if the state gets mucked up such that `banFromInstance` is false but `communityContext` is nil--safer to do something like this:
```suggestion
        if banFromInstance {
            instanceBan()
        } else if let communityContext {
            communityBan(from: communityContext)
        } else {
            assertionFailure("banFromInstance false but communityContext nil!")
        }
    }
```